python/segmentation/models/model_pspnet.py

```diff
@@ -0,0 +1,119 @@
+# DL library imports
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchvision.models import resnet50
+
+
+class auxiliary_branch(nn.Module):
+    """
+    Auxiliary branch helps in setting initial values for the residual blocks.
+    Auxiliary branch is only used during training and not during inference!
+    Auxiliary branch uses similar classifier and loss function of the main branch.
+    TotalLoss=α∗(auxloss)+(1−α)∗(mainloss)
+    Alpha is a hyperparameter. Authors used `alpha - 0.4*
+    """
+    def __init__(self, in_channels, num_classes):
+        super(auxiliary_branch, self).__init__()
+        self.aux = nn.Sequential(
+                    nn.Conv2d(in_channels , 256, kernel_size=3, padding=1, bias=False),
```

Comment:
Maube its not a time and place for it, but you could think about in future to not hardcode the kernel size but rather putting it a an input arguments to next have a possibility to experiment with the hyperparameters ;)

Reply:
All those parameters are based on that article: https://arxiv.org/pdf/1612.01105.pdf.